Lib/pkgutil.py

```diff
@@ -119,6 +119,9 @@ def iter_modules(path=None, prefix=''):
     """
     if path is None:
         importers = iter_importers()
+    elif isinstance(path, str) or isinstance(path, bytes):
```

Comment:
isinstance(path, bytes) is problematic.  I was hoping that walk_packages would throw some other error for bytes input (perhaps a TypeError on some string operation), because checking for all the bytes types is a bit of a pain.  And testing it, this is in fact true.  So the bytes check isn't needed, just the str check.  (Yes, the error is a bit mysterious, but if we want to address that, it requires more of a design discussion, and we shouldn't deal with that in this particular issue).